appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockVisibleExtension.java

```diff
@@ -0,0 +1,281 @@
+  public void onPropertyChange(String propertyName, String newValue) {
+    super.onPropertyChange(propertyName, newValue);
+    if (worker == null) return;
+    final JSONObject msg = new JSONObject();
+    msg.put("name", new JSONString(propertyName));
+    msg.put("value", convertPropertyValue(propertyName, newValue));
+    worker.postMessage(msg.toString());
+  }
+
+  private JSONValue convertPropertyValue(String name, String value) {
+    final String type = getProperties().getProperty(name).getEditorType();
+    switch (type) {
+      case PropertyTypeConstants.PROPERTY_TYPE_INTEGER:
+      case PropertyTypeConstants.PROPERTY_TYPE_NON_NEGATIVE_INTEGER:
+        return new JSONNumber(Integer.parseInt(value));
+      case PropertyTypeConstants.PROPERTY_TYPE_FLOAT:
+      case PropertyTypeConstants.PROPERTY_TYPE_NON_NEGATIVE_FLOAT:
+        return new JSONNumber(Float.parseFloat(value));
+      case PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN:
+      case PropertyTypeConstants.PROPERTY_TYPE_VISIBILITY:
+        return JSONBoolean.getInstance(Boolean.parseBoolean(value));
+      case PropertyTypeConstants.PROPERTY_TYPE_COLOR:
+        {
+          final String alpha = value.substring(2, 4);
+          final String baseHex = value.substring(4);
+          return new JSONString("#" + baseHex + alpha);
+        }
+      case PropertyTypeConstants.PROPERTY_TYPE_ASSET:
+        {
+          final String url = MockComponentsUtil.convertAssetValueToUrl(editor, value);
+          if (url == null) return null;
+          return new JSONString(url);
+        }
+      case PropertyTypeConstants.PROPERTY_TYPE_LENGTH:
+        {
+          final int intVal = Integer.parseInt(value);
+          if (intVal <= LENGTH_PERCENT_TAG) {
+            return new JSONString(Math.abs(intVal + 1000) + "%");
+          } else if (intVal == LENGTH_PREFERRED) {
+            return new JSONString("auto");
+          } else if (intVal == LENGTH_FILL_PARENT) {
+            return new JSONString("100%");
+          } else {
+            return new JSONString(intVal + "px");
+          }
+        }
+      case PropertyTypeConstants.PROPERTY_TYPE_TYPEFACE:
+        {
+          try {
+            final int intVal = Integer.parseInt(value);
+            if (intVal <= 1) {
+              return new JSONString("sans-serif");
+            } else if (intVal == 2) {
+              return new JSONString("serif");
+            } else if (intVal == 3) {
+              return new JSONString("monospace");
+            }
+          } catch (NumberFormatException e) {
+            final String typeface = value.substring(0, value.lastIndexOf("."));
+            if (Document.get().getElementById(typeface) == null) {
+              final String url = MockComponentsUtil.convertAssetValueToUrl(editor, value);
+              MockComponentsUtil.createFontResource(typeface, url, typeface);
+            }
+            return new JSONString(typeface);
+          }
+        }
+      default:
```

Comment:
Is this `default` statement reachable, or did we cover all types already?

Reply:
There are a whole lot of miscellaneous types in `PropertyTypeConstants` when this `default` will hit. It will also hit in case of text/string type.